Adding testing documentation

manolo opened the pull request:
Fixes #7

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/angular2-polymer/14)

<!-- Reviewable:end -->

manolo committed:
Adding Angular-CLI documentation.

Fixes #7 #15

platosha commented:
The instructions work for me, but got this error when trying to run `ng test` by the end:

```
Missing error handler on `socket`.
TypeError: (msg || "").replace is not a function
    at my-project/node_modules/karma/lib/reporter.js:45:23
```

<details><summary>

<i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>



> #### Adding testing documentation
> 
> Fixes #7
> 
> </details>

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 13 unresolved discussions.

---

_[docs/ng-cli.adoc, line 11 [r2]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImGxdpy5cbxxvIM8uY:-KImGxdpy5cbxxvIM8uZ:-1411678489) ([raw file](https://github.com/vaadin/angular2-polymer/blob/b15f73bafdbfcf659501d64e1a45031bf539810a/docs/ng-cli.adoc#L11)):_

> ``` AsciiDoc
> It is not only an scaffolding tool for creating and modifying a project, but also to accomplish other actions like building, running, debugging, and testing the project.
> 
> In this document we will describe all the modifications you have to make to your Angular-CLI project to use any polymer element.
> ```

Polymer is a proper name, needs to be capitalized. Note that there are other mentions of Polymer in this document.

---

_[docs/ng-cli.adoc, line 25 [r2]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImEytKyvYY7eDm3WNv:-KImEytKyvYY7eDm3WNw:2085526659) ([raw file](https://github.com/vaadin/angular2-polymer/blob/b15f73bafdbfcf659501d64e1a45031bf539810a/docs/ng-cli.adoc#L25)):_

> ``` AsciiDoc
> ----
> 
> Check that everything works, compiling, testing and running your project:
> ```

Missing conjunction after “works”. May be: “Check that everything works by compiling, testing, and running your project:”?

---

_[docs/ng-cli.adoc, line 57 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImPpBn4fREMIA0cy4K:-KImPpBooC7iF6sGqpkI:390813307) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L57)):_

> ``` AsciiDoc
> Now, you can install all the polymer elements that you need in your application.
> 
> For instance, to install all the elements in the https://elements.polymer-project.org/browse?package=paper-elements[Polymer Paper] collection, and the [element-name]#https://vaadin.com/elements/-/element/vaadin-combo-box[vaadin-combo-box]# element, run the following:
> ```

Should it be `[elementname]` instead of `[element-name]`?

---

_[docs/ng-cli.adoc, line 99 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImQeKTlHV2QSPm9-rR:-KImQeKTlHV2QSPm9-rS:-523485070) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L99)):_

> ``` AsciiDoc
> 
> [NOTE]
> The [filename]#webcomponents.js# polyfill is not necessary for browsers that fully implement the Web Components Spec (Only Chrome does so far).
> ```

Would prefer , `like Chrome`, or `i.e., Chrome`. That would be more accurate and would require less maintenance in future.

There is also Opera and some less-known Chromium-based browsers on the market.

---

_[docs/ng-cli.adoc, line 101 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImSi3H64KX8N-bdYN2:-KImSi3H64KX8N-bdYN3:1752932223) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L101)):_

> ``` AsciiDoc
> The [filename]#webcomponents.js# polyfill is not necessary for browsers that fully implement the Web Components Spec (Only Chrome does so far).
> 
> [NOTE]
> ```

Two consecutive note admonition blocks look strange. I would change a type of one or otherwise separate them a bit. Change the previous block to a tip, maybe?

---

_[docs/ng-cli.adoc, line 102 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImRkQxxTbYXC9t1j0H:-KImRkQxxTbYXC9t1j0I:-1854847437) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L102)):_

> ``` AsciiDoc
> 
> [NOTE]
> Although we can add polymer dependencies to the [filename]#index.html# file, we do prefer to maintain the auxiliary [filename]#elements.html#, so as we can reuse in tests as is described below.
> ```

“the auxiliary [filename]#elements.html# file”, I think.

Object is missing in the last clause: "so as we can reuse (what?) in tests...".

---

_[docs/ng-cli.adoc, line 107 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImTkjE_jGhX3bj_lES:-KImTkjE_jGhX3bj_lET:276083756) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L107)):_

> ``` AsciiDoc
> == Adding The Polymer Elements Directive
> 
> For using Polymer elements in Angular we need to import the [classname]#PolymerElement# directive from https://github.com/vaadin/angular2-polymer[@vaadin/angular2-polymer]. Thus we need to install the dependency by typing:
> ```

“in the Angular 2 application”

---

_[docs/ng-cli.adoc, line 128 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImWp5F3n-4Y_YUhfZ7:-KImWp5G7VlT_nwQXEAZ:847157024) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L128)):_

> ``` AsciiDoc
> ----
> 
> And make `ng` builder include the directive package by modifying the [propertyname]#vendorNpmFiles# property in the [filename]#angular-cli-build.js# file.
> ```

“make _the_ `ng` builder _to_ include the directive”

---

_[docs/ng-cli.adoc, line 149 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImWElIwesgjs9XiGJz:-KImWElIwesgjs9XiGK-:-1705121429) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L149)):_

> ``` AsciiDoc
> .src/app/my-project.component.html
> ----
> <div class="fit">
> ```

“fit” and “flex” are iron-flex-layout classes. Though this works for now, it's sort of discouraged to use without an explicit import of iron-flex-l

---

_[docs/ng-cli.adoc, line 176 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImUkDjSJdjQ-lv6WNP:-KImUkDjSJdjQ-lv6WNQ:232641129) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L176)):_

> ``` AsciiDoc
>     PolymerElement('paper-toolbar'),
>     PolymerElement('vaadin-combo-box'),
>     PolymerElement('paper-input'),
> ```

Unnecessary comma at the end of the line.

---

_[docs/ng-cli.adoc, line 190 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImUxeizvNSK0R95soE:-KImUxeizvNSK0R95soF:1484701741) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L190)):_

> ``` AsciiDoc
> 
> [source,html]
> .src/index.html
> ```

Where exactly in the index.html should I put this?

---

_[docs/ng-cli.adoc, line 194 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImVmEoUrMeDUbrAFnp:-KImVmEoUrMeDUbrAFnq:-1217260084) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L194)):_

> ``` AsciiDoc
> <style is="custom-style">
>   body {
>     @apply(--layout-fullbleed);
> ```

This effectively removes the body margin as well, which is good for the toolbar, but not the other elements in the component.

Should we drop the toolbar, this fullbleed style and the fit classes in this doc, maybe?

---

_[docs/ng-cli.adoc, line 213 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImXZzDH1dHtLJTptd0:-KImXZzDH1dHtLJTptd1:-1739607780) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L213)):_

> ``` AsciiDoc
> == Testing Polymer Elements
> 
> Angular 2 comes with a https://angular.io/docs/ts/latest/testing/[testing library] based on the https://karma-runner.github.io[karma.js] test runner.
> ```

The correct link, as it seems for me, is: https://angular.io/docs/ts/latest/guide/testing.html

Technically, the Angular 2 testing guide is based on Jasmine, and Karma runner is kin optional and not even documented there yet. I would rephrase it to say, that Angular-CLI projects come with Karma tests.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-:-KImWeV0y8r8G9_0TbR-:-147283171)_

<!-- Sent from Reviewable.io -->

manolo committed:
Applying reviewer suggestions

manolo commented:
Review status: 0 of 1 files reviewed at latest revision, 13 unresolved discussions.

---

_[docs/ng-cli.adoc, line 11 [r2]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImGxdpy5cbxxvIM8uY:-KIyS6WuMhylDC32KVMC:-499146009) ([raw file](https://github.com/vaadin/angular2-polymer/blob/b15f73bafdbfcf659501d64e1a45031bf539810a/docs/ng-cli.adoc#L11)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> Polymer is a proper name, needs to be capitalized. Note that there are other mentions of Polymer in this document.
> </details>
> Done.

---

_[docs/ng-cli.adoc, line 25 [r2]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImEytKyvYY7eDm3WNv:-KIySGrQrGdspbyBPfWo:-499146009) ([raw file](https://github.com/vaadin/angular2-polymer/blob/b15f73bafdbfcf659501d64e1a45031bf539810a/docs/ng-cli.adoc#L25)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> Missing conjunction after “works”. May be: “Check that everything works by compiling, testing, and running your project:”?
> </details>
> Done.

---

_[docs/ng-cli.adoc, line 57 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImPpBn4fREMIA0cy4K:-KIySPGtT35KUfqZz20w:-499146009) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L57)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> Should it be `[elementname]` instead of `[element-name]`?
> </details>
> Done.

---

_[docs/ng-cli.adoc, line 99 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImQeKTlHV2QSPm9-rR:-KIyScZu_CyGA6IKOINU:-499146009) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L99)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> Would prefer , `like Chrome`, or `i.e., Chrome`. That would be more accurate and would require less maintenance in future.
> 
> There is also Opera and some less-known Chromium-based browsers on the market.
> </details>
> Done.

---

_[docs/ng-cli.adoc, line 101 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImSi3H64KX8N-bdYN2:-KIySrj3MeK3nA_5W7za:-499146009) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L101)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> Two consecutive note admonition blocks look strange. I would change a type of one or otherwise separate them a bit. Change the previous block to a tip, maybe?
> </details>
> Done.

---

_[docs/ng-cli.adoc, line 102 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImRkQxxTbYXC9t1j0H:-KIyT5LyFjPnH71oAVez:-499146009) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L102)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> “the auxiliary [filename]#elements.html# file”, I think.
> 
> Object is missing in the last clause: "so as we can reuse (what?) in tests...".
> </details>
> Done.

---

_[docs/ng-cli.adoc, line 107 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImTkjE_jGhX3bj_lES:-KIyXPrcjGdGUrX9_GDB:-499146009) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L107)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> “in the Angular 2 application”
> </details>
> Done.

---

_[docs/ng-cli.adoc, line 128 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImWp5F3n-4Y_YUhfZ7:-KIyTLK6OTUNAHWfAEms:396221706) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L128)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> “make _the_ `ng` builder _to_ include the directive”
> </details>
> make is a special case that is not followed by 'to':

```
Verbs followed by an infinitive without to
Let, make
Let and make are followed by an infinitive without to in active voice sentences. They always have an object (underlined) before the infinitive:

Let me show you this DVD I’ve got.

They made us wait while they checked our documents.

Not: They made us to wait …
```

---

_[docs/ng-cli.adoc, line 149 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImWElIwesgjs9XiGJz:-KIyVBLJDYdHK14U84R3:1234089427) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L149)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> “fit” and “flex” are iron-flex-layout classes. Though this works for now, it's sort of discouraged to use without an explicit import of iron-flex-l
> </details>
> Well paper-styles is a convenience bundle to  import flex, colors, fonts etc. I think we shouldn't be redundant importing those elements. 

---

_[docs/ng-cli.adoc, line 176 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImUkDjSJdjQ-lv6WNP:-KIyVmqyOq8Sy3q0S7Hp:-499146009) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L176)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> Unnecessary comma at the end of the line.
> </details>
> Done.

---

_[docs/ng-cli.adoc, line 190 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImUxeizvNSK0R95soE:-KIyW7a4vr6fCqIcez3M:-499146009) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L190)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> Where exactly in the index.html should I put this?
> </details>
> Done.

---

_[docs/ng-cli.adoc, line 194 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImVmEoUrMeDUbrAFnp:-KIyW9FUFnrTIVpiHr30:833741900) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L194)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> This effectively removes the body margin as well, which is good for the toolbar, but not the other elements in the component.
> 
> Should we drop the toolbar, this fullbleed style and the fit classes in this doc, maybe?
> </details>
> Well the demo is just an example not a template, the tutorial is much more complete for this. 
> I dont think is relevant whether the components have margin, but that the components are shown.

---

_[docs/ng-cli.adoc, line 213 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImXZzDH1dHtLJTptd0:-KIyXQbZvRmhzJKq9MAU:-499146009) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L213)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> The correct link, as it seems for me, is: https://angular.io/docs/ts/latest/guide/testing.html
> 
> Technically, the Angular 2 testing guide is based on Jasmine, and Karma runner is kin optional and not even documented there yet. I would rephrase it to say, that Angular-CLI projects come with Karma tests.
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14)_

<!-- Sent from Reviewable.io -->

platosha commented:
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

_[docs/ng-cli.adoc, line 128 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImWp5F3n-4Y_YUhfZ7:-KJ-WAE9uWmypI35uJbi:736671256) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L128)):_
<details><summary>

<i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>



> make is a special case that is not followed by 'to':
> 
> ```
> Verbs followed by an infinitive without to
> Let, make
> Let and make are followed by an infinitive without to in active voice sentences. They always have an object (underlined) before the infinitive:
> 
> Let me show you this DVD I’ve got.
> 
> They made us wait while they checked our documents.
> 
> Not: They made us to wait …
> ```
> 
> </details>
> Alright. But what about the article: “_the_ `ng` builder”?

---

_[docs/ng-cli.adoc, line 149 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImWElIwesgjs9XiGJz:-KJ-XKPrKjbRSLL3v5GO:-1256469833) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L149)):_
<details><summary>

<i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>



> Well paper-styles is a convenience bundle to  import flex, colors, fonts etc. I think we shouldn't be redundant importing those elements. 
> </details>
> Note that `paper-styles` imports `iron-flex-layout/classes/iron-flex-layout.html`, which is depricated and gives a console warning.

---

_[docs/ng-cli.adoc, line 194 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImVmEoUrMeDUbrAFnp:-KJ-YDRXTfs9zfwnnzQX:1191429606) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L194)):_
<details><summary>

<i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>



> Well the demo is just an example not a template, the tutorial is much more complete for this. 
> I dont think is relevant whether the components have margin, but that the components are shown.
> </details>
> Right. For this exact reason it’s irrelevant to make the body fullbleed.

If we would just remove the fullbleed from body and skip paper-toolbar, then:
- The example will become more simple.
- We’ll avoid dependency from the deprecated iron-flex-layout classes, imported by paper-styles, like `fit` and `flex`.
- The elements that remain in the example will look better with the body margin, then they look now without the body margin. It will make a better impression overall.

---

_[docs/ng-cli.adoc, line 8 [r4]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KJ-VJOh1GggjlcEGk52:-KJ-VJOizoVMJFe9JnL-:-240014981) ([raw file](https://github.com/vaadin/angular2-polymer/blob/ced8134b5ae5f3cbe16410794f25ab940fb2ffe6/docs/ng-cli.adoc#L8)):_

> ``` AsciiDoc
> == Introduction
> 
> https://github.com/angular/angular-cli[Angular-CLI] is a command line tool for Angular 2. It is not only an scaffolding tool for creating and modifying a project, but also for other actions like building, running, debugging, and testing the project.
> ```

“a scaffolding tool”

---

_[docs/ng-cli.adoc, line 55 [r4]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KJ-VGWvlyE77FtwEPzj:-KJ-VGWvlyE77FtwEPzk:-1958730476) ([raw file](https://github.com/vaadin/angular2-polymer/blob/ced8134b5ae5f3cbe16410794f25ab940fb2ffe6/docs/ng-cli.adoc#L55)):_

> ``` AsciiDoc
> ----
> 
> Now, you can install all the polymer elements that you need in your application.
> ```

Capitalize “Polymer”, please.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14)_

<!-- Sent from Reviewable.io -->

platosha commented:
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

_[docs/ng-cli.adoc, line 61 [r4]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KJ-lx2FXEgahsxAx_WA:-KJ-lx2GESSgQqYirjkr:1976403979) ([raw file](https://github.com/vaadin/angular2-polymer/blob/ced8134b5ae5f3cbe16410794f25ab940fb2ffe6/docs/ng-cli.adoc#L61)):_

> ``` AsciiDoc
> [subs="normal"]
> ----
> [prompt]#$# [command]#npm# install bower -g
> ```

Duplicate command `npm install bower -g`. Bower was already installed a few lines above.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14)_

<!-- Sent from Reviewable.io -->

platosha commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

_[docs/ng-cli.adoc, line 83 [r4]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KJ-nIteJVE-wqnfeSKX:-KJ-nItfwvJJCc6WR3SX:-1591514282) ([raw file](https://github.com/vaadin/angular2-polymer/blob/ced8134b5ae5f3cbe16410794f25ab940fb2ffe6/docs/ng-cli.adoc#L83)):_

> ``` AsciiDoc
> ----
> <body>
>   ...
> ```

I think, we should be more concrete here about what lines to leave and what to replace inside the `<body>`.

While following this step, I’ve just removed this line:

```
    {{#each scripts.polyfills}}<script src="{{.}}"></script>{{/each}}
```

 by mistake.

Maybe list the complete `<body>` section code here? Otherwise, these kind of misunderstandings will break the application.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14)_

<!-- Sent from Reviewable.io -->

manolo committed:
Removing paper-toolbar example

manolo commented:
Review status: 0 of 1 files reviewed at latest revision, 7 unresolved discussions.

---

_[docs/ng-cli.adoc, line 128 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImWp5F3n-4Y_YUhfZ7:-KJ-rtFPrJ0P888IveuM:-499146009) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L128)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> Alright. But what about the article: “_the_ `ng` builder”?
> </details>
> Done.

---

_[docs/ng-cli.adoc, line 149 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImWElIwesgjs9XiGJz:-KJ-tAYzmPFbgPzzHqHN:-499146009) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L149)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> Note that `paper-styles` imports `iron-flex-layout/classes/iron-flex-layout.html`, which is depricated and gives a console warning.
> </details>
> Done.

---

_[docs/ng-cli.adoc, line 194 [r3]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KImVmEoUrMeDUbrAFnp:-KJ-tEoB_FWoJcqcCrL_:1293114140) ([raw file](https://github.com/vaadin/angular2-polymer/blob/665d31667874af1fd08360cd7f1aaf776acc1b3b/docs/ng-cli.adoc#L194)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> Right. For this exact reason it’s irrelevant to make the body fullbleed.
> 
> If we would just remove the fullbleed from body and skip paper-toolbar, then:
> - The example will become more simple.
> - We’ll avoid dependency from the deprecated iron-flex-layout classes, imported by paper-styles, like `fit` and `flex`.
> - The elements that remain in the example will look better with the body margin, then they look now without the body margin. It will make a better impression overall.
>   </details>
>   Done. Removed the component, as well as the fullbleed. I agree that including the component does not give any point to the user.

---

_[docs/ng-cli.adoc, line 8 [r4]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KJ-VJOh1GggjlcEGk52:-KJ-r3Gm3GNobf3fV5ZD:-499146009) ([raw file](https://github.com/vaadin/angular2-polymer/blob/ced8134b5ae5f3cbe16410794f25ab940fb2ffe6/docs/ng-cli.adoc#L8)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> “a scaffolding tool”
> </details>
> Done.

---

_[docs/ng-cli.adoc, line 55 [r4]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KJ-VGWvlyE77FtwEPzj:-KJ-tVOcYyRgsRo5ecji:-499146009) ([raw file](https://github.com/vaadin/angular2-polymer/blob/ced8134b5ae5f3cbe16410794f25ab940fb2ffe6/docs/ng-cli.adoc#L55)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> Capitalize “Polymer”, please.
> </details>
> Done.

---

_[docs/ng-cli.adoc, line 61 [r4]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KJ-lx2FXEgahsxAx_WA:-KJ-r5ljA6ahU2EdWUkD:-499146009) ([raw file](https://github.com/vaadin/angular2-polymer/blob/ced8134b5ae5f3cbe16410794f25ab940fb2ffe6/docs/ng-cli.adoc#L61)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> Duplicate command `npm install bower -g`. Bower was already installed a few lines above.
> </details>
> Done.

---

_[docs/ng-cli.adoc, line 83 [r4]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KJ-nIteJVE-wqnfeSKX:-KJ-rl8d28X8kKQTdT1L:-499146009) ([raw file](https://github.com/vaadin/angular2-polymer/blob/ced8134b5ae5f3cbe16410794f25ab940fb2ffe6/docs/ng-cli.adoc#L83)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> I think, we should be more concrete here about what lines to leave and what to replace inside the `<body>`.
> 
> While following this step, I’ve just removed this line:
> 
> ```
>     {{#each scripts.polyfills}}<script src="{{.}}"></script>{{/each}}
> ```
> 
>  by mistake.
> 
> Maybe list the complete `<body>` section code here? Otherwise, these kind of misunderstandings will break the application.
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14)_

<!-- Sent from Reviewable.io -->

platosha commented:
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

_[docs/ng-cli.adoc, line 61 [r4]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KJ-lx2FXEgahsxAx_WA:-KJ0xpSsh8qVNlAqLvrK:402788713) ([raw file](https://github.com/vaadin/angular2-polymer/blob/ced8134b5ae5f3cbe16410794f25ab940fb2ffe6/docs/ng-cli.adoc#L61)):_
<details><summary>

<i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>



> Done.
> </details>
> Ahem, the command is still here. What has changed?

---

_[docs/ng-cli.adoc, line 191 [r5]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KJ0yhogdLN0lgWSZ_bB:-KJ0yhogdLN0lgWSZ_bC:-617895823) ([raw file](https://github.com/vaadin/angular2-polymer/blob/2c6a71c7e3c15bd6ac684a84172b6c8214558962/docs/ng-cli.adoc#L191)):_

> ``` AsciiDoc
>     body {
>       @apply(--paper-font-body1);
>       font-family: 'Roboto', 'Noto', sans-serif;
> ```

In case if you don’t want a `paper-styles` import in the document, would be cleaner to avoid using its mixins and variables.

Previously, when we had `paper-styles` imported, this was fine to use the mixin. Now that you removed the import, the mixin depends on a side-effect of importing paper-styled elements.

Let us, maybe, bring the `paper-styles` import back? This would make the font-family style rule unnecessary here, we should probably remove it.

I was not against the `paper-styles` import in general. Just wanted to point you, that the iron-flex-layout classes, that `paper-styles` depend on, are not good to rely on.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14)_

<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

_[docs/ng-cli.adoc, line 61 [r4]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KJ-lx2FXEgahsxAx_WA:-KJ4wLs1vzTcqZdd0K1a:-499146009) ([raw file](https://github.com/vaadin/angular2-polymer/blob/ced8134b5ae5f3cbe16410794f25ab940fb2ffe6/docs/ng-cli.adoc#L61)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> Ahem, the command is still here. What has changed?
> </details>
> Done.

---

_[docs/ng-cli.adoc, line 191 [r5]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KJ0yhogdLN0lgWSZ_bB:-KJ4wKNda5myK80NSX6U:-499146009) ([raw file](https://github.com/vaadin/angular2-polymer/blob/2c6a71c7e3c15bd6ac684a84172b6c8214558962/docs/ng-cli.adoc#L191)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> In case if you don’t want a `paper-styles` import in the document, would be cleaner to avoid using its mixins and variables.
> 
> Previously, when we had `paper-styles` imported, this was fine to use the mixin. Now that you removed the import, the mixin depends on a side-effect of importing paper-styled elements.
> 
> Let us, maybe, bring the `paper-styles` import back? This would make the font-family style rule unnecessary here, we should probably remove it.
> 
> I was not against the `paper-styles` import in general. Just wanted to point you, that the iron-flex-layout classes, that `paper-styles` depend on, are not good to rely on.
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14)_

<!-- Sent from Reviewable.io -->

platosha commented:
Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

_[docs/ng-cli.adoc, line 191 [r6]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KJ5FgjQJqUB0d9GZNx-:-KJ5FgjQJqUB0d9GZNx0:1555424236) ([raw file](https://github.com/vaadin/angular2-polymer/blob/5f879cb4afbe7c85bc78e1fd866c9f940be5240e/docs/ng-cli.adoc#L191)):_

> ``` AsciiDoc
>   <style is="custom-style">
>     body {
>       @apply(--paper-font-body1);
> ```

This mixin is defined in the `paper-styles/typography.html` file, not in the default theme.

---

_[docs/ng-cli.adoc, line 192 [r6]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KJ5Fk9qEnvFg-lIr5aS:-KJ5Fk9qEnvFg-lIr5aT:406087382) ([raw file](https://github.com/vaadin/angular2-polymer/blob/5f879cb4afbe7c85bc78e1fd866c9f940be5240e/docs/ng-cli.adoc#L192)):_

> ``` AsciiDoc
>     body {
>       @apply(--paper-font-body1);
>       font-family: 'Roboto', 'Noto', sans-serif;
> ```

This line is redundant, the same is declared in the `--paper-font-body1` mixin.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14)_

<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

_[docs/ng-cli.adoc, line 191 [r6]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KJ5FgjQJqUB0d9GZNx-:-KJ5v3TRta9AFyY1LJR0:-499146009) ([raw file](https://github.com/vaadin/angular2-polymer/blob/5f879cb4afbe7c85bc78e1fd866c9f940be5240e/docs/ng-cli.adoc#L191)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> This mixin is defined in the `paper-styles/typography.html` file, not in the default theme.
> </details>
> Done.

---

_[docs/ng-cli.adoc, line 192 [r6]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-KJ5Fk9qEnvFg-lIr5aS:-KJ5v3sEtEzQeIgk8STJ:-499146009) ([raw file](https://github.com/vaadin/angular2-polymer/blob/5f879cb4afbe7c85bc78e1fd866c9f940be5240e/docs/ng-cli.adoc#L192)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> This line is redundant, the same is declared in the `--paper-font-body1` mixin.
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14)_

<!-- Sent from Reviewable.io -->

platosha commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> The instructions work for me, but got this error when trying to run `ng test` by the end:
> 
> ```
> Missing error handler on `socket`.
> TypeError: (msg || "").replace is not a function
>     at /Users/antonplatonov/Work/2016/my-project/node_modules/karma/lib/reporter.js:45:23
> ```
> 
> </details>

---

Reviewed 1 of 1 files at r7.
Review status: all files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14#-:-KJ64SwXMYdLiVnmEEsA:1417088721)_

<!-- Sent from Reviewable.io -->

manolo committed:
Adding metainfo for docsite

platosha commented:
Reviewed 1 of 1 files at r8.
Review status: all files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/angular2-polymer/14)_

<!-- Sent from Reviewable.io -->
